### PR TITLE
Enable CSRF Protection protect

### PR DESCRIPTION
**Pull request by rhythmdesai404**
1. Enabled CSRF Protection

### CODE REVIEW

**Comment by ericmustin** [comment]
Misclick, sorry about that. Is this PR ready for review?

File: instrumentation/action_pack/test/test_helpers/controllers/example_controller.rb [hunk 1]
```diff
@@ -7,6 +7,9 @@
 class ExampleController < ActionController::Base
   include ::Rails.application.routes.url_helpers
 
+  # Enable CSRF Protection
+  protect_from_forgery with: :exception
```
Comment (kaylareopelle) [inline instrumentation/action_pack/test/test_helpers/controllers/example_controller.rb:11]: Hi @rhythmdesai404! Since I haven't heard back from you after Eric's comment earlier, I'm assuming this is ready for review. 

I'm curious about why we need to turn this on. This is test code, and shouldn't impact the user. Why do we need to enable it here?

Comment (rhythmdesai404) [reply instrumentation/action_pack/test/test_helpers/controllers/example_controller.rb:11]: Hi @kaylareopelle! Great question. Even though it's test code, enabling CSRF protection is important because it helps ensure that the tests accurately reflect the security measures in place for the actual application. By turning it on, we:
Simulate Real Conditions: Make sure the test environment mimics the real application environment, including its security features.
Consistency: Maintain consistency in how the application handles requests, whether in testing or in production.
In short, enabling CSRF protection in test code helps ensure that your application is secure and behaves as expected in all scenarios.

**Review by rhythmdesai404 — COMMENTED** [review]
# Enable CSRF protection to ensure test environment security
# This change simulates real application conditions, catches potential issues early,
# and maintains consistency in request handling across all environments.
protect_from_forgery with: :exception

File: instrumentation/action_pack/test/test_helpers/controllers/example_controller.rb (hunk 1 above)
Comment (kaylareopelle) [reply instrumentation/action_pack/test/test_helpers/controllers/example_controller.rb:11]: Thank you for the context, @rhythmdesai404! That makes sense to me. If you'd still like to see this change, please reopen the PR and we'll give it a look.
